src/main/java/codechicken/translocators/init/DataGenerators.java

```diff
@@ -0,0 +1,64 @@
+package codechicken.translocators.init;
+
+import codechicken.lib.datagen.ItemModelProvider;
+import net.minecraft.data.DataGenerator;
+import net.minecraft.data.ItemTagsProvider;
+import net.minecraft.tags.ItemTags;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.client.model.generators.ExistingFileHelper;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.event.lifecycle.GatherDataEvent;
+
+import static codechicken.translocators.Translocators.MOD_ID;
+
+@Mod.EventBusSubscriber (modid = MOD_ID, bus = Mod.EventBusSubscriber.Bus.MOD)
+public class DataGenerators {
+
+    @SubscribeEvent
+    public static void gatherDataGenerators(GatherDataEvent event) {
+        DataGenerator gen = event.getGenerator();
+        ExistingFileHelper files = event.getExistingFileHelper();
+        if (event.includeClient()) {
+            gen.addProvider(new ItemModels(gen, files));
+        }
+        gen.addProvider(new ItemTags(gen));
+    }
+
+    private static class ItemModels extends ItemModelProvider {
+
+        public ItemModels(DataGenerator generator, ExistingFileHelper existingFileHelper) {
+            super(generator, MOD_ID, existingFileHelper);
+        }
+
+        @Override
+        protected void registerModels() {
+            noTexture(ModContent.itemTranslocatorItem);
+            noTexture(ModContent.fluidTranslocatorItem);
+            generated(ModContent.diamondNuggetItem);
+        }
+
+        @Override
+        public String getName() {
+            return "Translocators Item models";
+        }
+    }
+
+    private static class ItemTags extends ItemTagsProvider {
+
+        public ItemTags(DataGenerator gen) {
+            super(gen);
+        }
+
+        @Override
+        protected void registerTags() {
+            getBuilder(ModContent.diamondNuggetTag).add(ModContent.diamondNuggetItem);
+            getBuilder(new net.minecraft.tags.ItemTags.Wrapper(new ResourceLocation("forge:nuggets"))).add(ModContent.diamondNuggetItem);
```

Comment:
From what I understood in [the Forge docs](https://mcforge.readthedocs.io/en/latest/utilities/tags/), I was able to find a [`forge:nuggets` tag](https://github.com/MinecraftForge/MinecraftForge/blob/1.15.x/src/generated/resources/data/forge/tags/items/nuggets.json).

I couldn't find a way of adding the item to this tag, so I tried with this. I'm guessing there's some king of enunm / container class somewhere, but it's not mentioned in the docs.

Reply:
Instead of creating a new tag here, just use the tag instance you created previously(in ModContent) for the Diamond Nugget.

Reply:
So the tag instance in `ModContent` (which I am moving to the `Translocators` class), is for `forge:nuggets/diamond`. This line right here is for `forge:nuggets` more broadly. But maybe I can use the same pattern for the `forge:nuggets` tag?

Reply:
We don't want the Diamond Nugget added to that tag though, we want it added to `forge:nuggets/diamond`.